File: API/Entities/Activity.cs
using System.Text.Json.Serialization;

namespace API.Entities
{
    public class Activity : BaseEntity
    {
        [JsonPropertyName("name")]   
        public string Name { get; set; }
        [JsonPropertyName("description")]   
        public string Description { get; set; }
        public int MaxSeats { get; set; }

        // Activity has period as foreign key
        [JsonPropertyName("periodId")]   
        public int PeriodId { get; set; }
        public Period Period { get; set; }
    }
}

File: react_client/src/app/models/activity.ts
import { Period } from "./period";

export interface Activity {
    id: number;
    name: string;
    description: string;
    maxSeats: number;
    periodId: number;
    period: Period;
}

export interface ActivityCreation {
    name: string;
    description: string;
    maxSeats: number;
    periodId: number;
}